### DISCUSSION TITLE
Problem finding entry point via importslib.metadata.entry_points in packaged app

### DISCUSSION THREAD

**Discussion by pearcetm** [pyinstaller/pyinstaller#8325] [Help] [answered]
I'm trying to package an application that uses a library ([large_image](https://github.com/girder/large_image)) which uses additional packages by iterating over `importslib.metadata.entry_points` to find installed plugins. However, in my packaged app, no plugins are being found.

I've added these plugins using the `--hidden-import` flag and tried to include their metadata using `--copy-metadata` and `--recursive-copy-metadata`, but the entry points still aren't being found.

Is there a trick to get this to work? A few years ago [this comment](https://github.com/pyinstaller/pyinstaller/issues/3050#issuecomment-347868735) indicated that such metadata wasn't included, and I haven't yet stumbled across anything official more recently about this.

**Comment by rokm** [comment, marked as answer]
There's a [`collect_entry_point`](https://pyinstaller.org/en/stable/hooks.html#PyInstaller.utils.hooks.collect_entry_point) hook utility function that you can use in your spec file (or a hook for `large_image`). It is not mapped to CLI, though (so there is no equivalent to `--hidden-import` or `--copy-metadata`.

**Comment by pearcetm** [comment]
Thanks for the fast response! Adding this to the `spec` file did the trick. If I can figure out some additional parts of this, I'll see about creating a hook for this package. Marking as solved. Getting this to work led to an additional issue, which I'll open separately.
